lib/talkable.rb

```diff
@@ -27,17 +28,27 @@ def visitor_uuid=(uuid)
       Thread.current[UUID] = uuid
     end
 
-    def with_uuid(uuid)
+    def find_or_generate_uuid
+      visitor_uuid || Talkable::API::Visitor.create[:uuid]
+    end
+
+    def current_url
```

Comment:
WDYT about storing `visitor_uuid` and `current_url` in one `Thread.current` variable?